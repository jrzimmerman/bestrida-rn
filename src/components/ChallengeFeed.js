import React from 'react'
import {
  Image,
  ListView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native'
import { styles } from './styles'

class ChallengeFeed extends React.Component {
  constructor (props) {
    super(props)
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})
    this.state = {
      dataSource: ds.cloneWithRows([
        'John', 'Joel', 'James', 'Jimmy', 'Jackson', 'Jillian', 'Julie', 'Devin'
      ])
    }
  }

  render () {
    return (
      <View style={styles.container}>
        <StatusBar barStyle="light-content" />
        <TouchableOpacity style={feedStyles.button}>
          <Text style={styles.buttonText}>Create Challenge</Text>
        </TouchableOpacity>
        <ListView
          style={feedStyles.list}
          dataSource={this.state.dataSource}
          renderRow={(rowData) => (
            <TouchableOpacity style={styles.row}>
              <View style={styles.challengeImageView}>
                <Image style={styles.challengeImage} source={require('../images/strava_profile_pic.png')} />
              </View>
              <View style={styles.challengeDetail}>
                <Text style={styles.challengeText}>Opponent: OPPONENT</Text>
                <Text style={styles.challengeText}>Segment: SEGMENT NAME</Text>
                <Text style={styles.challengeText}>Complete By: DATE HERE</Text>
              </View>
            </TouchableOpacity>
          )}
        />
      </View>
    )
  }
}

const feedStyles = StyleSheet.create({
  list: {
    alignSelf: 'stretch',
    marginTop: -60
  },
  button: {
    marginTop: 80,
    alignSelf: 'stretch',
    justifyContent: 'space-around',
    flexDirection: 'row',
    margin: 10,
    paddingVertical: 6,
    paddingHorizontal: 8,
    borderRadius: 4,
    height: 45,
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: '#ef473a'
  }
})

export default ChallengeFeed
