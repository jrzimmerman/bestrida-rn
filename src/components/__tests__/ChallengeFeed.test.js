import 'react-native';
import React from 'react';
import { ChallengeFeed } from '../ChallengeFeed';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders ChallengeFeed component', () => {
  const pending = {
    loading: false,
    challenges: [],
    error: null
  };
  const tree = renderer
    .create(<ChallengeFeed pending={pending} dispatch={fn => fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
