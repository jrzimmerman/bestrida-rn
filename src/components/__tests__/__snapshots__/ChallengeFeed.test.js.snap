// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChallengeFeed component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2B2B2B",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 0.15,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "transparent",
          "borderColor": "#ef473a",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around",
          "margin": 10,
          "marginTop": 20,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#ef473a",
            "flexDirection": "row",
            "fontSize": 14,
          }
        }
      >
        Create Challenge
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 0.85,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentInset={
        Object {
          "bottom": 55,
        }
      }
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      enableEmptySections={true}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
        }
      }
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
</View>
`;
